tests/python_package_test/test_dask.py

```diff
@@ -805,6 +808,67 @@ def f(part):
         assert 'foo' in str(info.value)
 
 
+@pytest.mark.parametrize('task,output', tasks_outputs)
+def test_training_having_workers_without_data(client, task, output):
+
+    def repartition(collection):
```

Comment:
```suggestion
    def repartition(collection):
        """Merge the parts of a Dask collection into a single partition"""
```

I love this function! But it has a very general name, so I'd like to recommend adding this docstring so others working on this code in the future will understand what its purpose is.